x-pack/plugins/enterprise_search/server/saved_objects/app_search/telemetry.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsType } from 'src/core/server';
+
+export const AS_TELEMETRY_NAME = 'app_search_telemetry';
+
+export interface ITelemetrySavedObject {
+  ui_viewed: {
+    setup_guide: number;
+    engines_overview: number;
+  };
+  ui_error: {
+    cannot_connect: number;
+    no_as_account: number;
+  };
+  ui_clicked: {
+    create_first_engine_button: number;
+    header_launch_button: number;
+    engine_table_link: number;
+  };
+}
+
+export const appSearchTelemetryType: SavedObjectsType = {
+  name: AS_TELEMETRY_NAME,
+  hidden: false,
+  namespaceType: 'single',
+  mappings: {
+    properties: {
```

Comment:
Probably none of these fields need to be indexed. See https://github.com/elastic/kibana/issues/43673 for more info.

Reply:
Oh, cool, thanks a ton Josh! I'll admit I definitely just copied this setup from the upgrade_assistant plugin 😅 Definitely none of this needs to be searched at all haha.

I'm not super familiar with the structure here but do I just add a single `index: false` below this line or for each `properties` blob? (Will test it out both ways!)

Reply:
Just to check - I've added `index: false` to each field but it sounds like (from reading the PR) it might be even better/more performant to just do
```
mappings: {
  type: 'object',
  enabled: false,
}
```
1. Is that indeed the "right" way to go?
2. Do I just `@ts-ignore` if I do so? (Typescript complains otherwise)